src/omotes_orchestrator/main.py

```diff
@@ -396,6 +401,40 @@ def job_cancellation_handler(self, job_cancellation: JobCancel) -> None:
                 )
                 self._cleanup_job(job_id)
 
+    def dead_lettered_job_result_handler(self, job_result: JobResult) -> None:
+        """Handle the received dead lettered job result.
+
+        When the log level is set at the DEBUG level, the dead lettered job result will be written
+        to a local file.
+
+        :param job_result: Job result message.
+        """
+        logger.info("Received a dead lettered job_%s_result with result type as: %s",
+                    job_result.uuid,
+                    job_result.result_type)
+
+        log_level = logger.getEffectiveLevel()
+        if log_level == logging.DEBUG:
```

Comment:
Log level may also be `NOT_SET`. We should check `log_level <= logging.DEBUG` (https://docs.python.org/3/library/logging.html#logging-levels)

Reply:
Good tips! Updated in 1a5bb3fe40a03f272badacea7bd8e413a3c8c4a5